app.js

```diff
@@ -97,33 +97,51 @@ controller.on('slash_command', (bot, message) => {
     let splitMsg = msg.split(' ');
     let pranked = splitMsg[1] ? splitMsg[1].replace('@', '').toLowerCase() : '';
 
-    if(splitMsg[0] === 'PRANK' && splitMsg[1]) {
-      if(!RobinBot.prankedUsers.hasOwnProperty(pranked)) {
-        reply.private = `Holy Prankster! I've added ${pranked} to my list!`;
-        RobinBot.prankedUsers[pranked] = {
-          lastPrankTime: 0,
-          nextPrankTime: 0,
-          prankerID: message.user,
-          lastPrankMessage: ''
-        };
-      } else {
-        reply.private = `Holy Confusion! ${pranked} is already on my list!`;
+    if(pranked) {
+      switch (splitMsg[0]) {
+      case 'PRANK': {
+        if(!RobinBot.prankedUsers.hasOwnProperty(pranked)) {
+          reply.private = `Holy Prankster! I've added ${pranked} to my list!`;
+          RobinBot.prankedUsers[pranked] = {
+            lastPrankTime: 0,
+            nextPrankTime: 0,
+            prankerID: message.user,
+            lastPrankMessage: ''
+          };
+        } else {
+          reply.private = `Holy Confusion! ${pranked} is already on my list!`;
+        }
       }
-
-    } else if(splitMsg[0] === 'FORGIVE' && splitMsg[1]) {
-      if(RobinBot.prankedUsers.hasOwnProperty(pranked)) {
-        if(message.user === RobinBot.prankedUsers[pranked].prankerID) {
-          reply.private = `Holy Prankster! ${pranked} has been removed from my list!`;
-          delete RobinBot.prankedUsers[pranked];
+        break;
+      case 'FORGIVE': {
+        if(RobinBot.prankedUsers.hasOwnProperty(pranked)) {
+          if(message.user === RobinBot.prankedUsers[pranked].prankerID) {
+            reply.private = `Holy Prankster! ${pranked} has been removed from my list!`;
+            delete RobinBot.prankedUsers[pranked];
+          } else {
+            reply.private = 'Holy Prankster! Only the person that pranked you can remove you from my list!';
+          }
+        } else if(pranked === 'all') {
+          let forgiveAllReply = 'Holy Holiness! Users removed from the prank list: \n';
+          console.log('ALL_PRANKED', RobinBot.prankedUsers);
+          for(let [prankedName, prankedObject] of Object.entries(RobinBot.prankedUsers)) {
+            if(message.user === prankedObject.prankerID) {
+              forgiveAllReply += prankedName + ' \n';
```

Comment:
I think this is better off in a single line. You can create an empty array above `let forgiveAllReply` and then push into it all the names that are being removed. Then you can append the `forgiveAllReply` with `arrayOfPrankedUsersRemoved.join(',')`.